ggml-kompute.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "ggml.h"
+#include "ggml-backend.h"
+
+#include <cstddef>
+#include <string>
+#include <vector>
+
+struct ggml_vk_device {
+    int index = 0;
+    int type = 0;           // same as VkPhysicalDeviceType
+    size_t heapSize = 0;
+    std::string name;
+    std::string vendor;
+    int subgroupSize = 0;
+};
+
+std::vector<ggml_vk_device> ggml_vk_available_devices(size_t memoryRequired);
+bool ggml_vk_init_device(size_t memoryRequired, const std::string &device);
+bool ggml_vk_init_device(const ggml_vk_device &device);
+bool ggml_vk_init_device(int device);
+bool ggml_vk_free_device();
+bool ggml_vk_has_vulkan();
+bool ggml_vk_has_device();
+bool ggml_vk_using_vulkan();
+ggml_vk_device ggml_vk_current_device();
```

Comment:
Ideally this should be removed, or the API adapted to C, but if this is still absolutely necessary for GPT4All, enclose in `#ifdef __cplusplus` so that the basic ggml-backend API can still be used from C code.